### PR TITLE
`main` and `map-implementation` merge

### PR DESCRIPTION
**Pull request by Turtyo**
# Description

This does the merge between main and map.

## Related issue(s)

None

## List of changes

- Resolve initial conflicts
- Add typing
- Rework some tests (@Tysterman74 would like your approval on the `test_map_ui`)

## Additional notes

`map-implementation` shouldn't be used going forward, as it was only used for initial development of the map.

### CODE REVIEW

File: #Scenes/Map31B5.tmp [hunk 1]
```diff
@@ -0,0 +1,89 @@
+[gd_scene load_steps=4 format=3 uid="uid://ki060lfrgty4"]
```
Comment (Tysterman74) [inline #Scenes/Map31B5.tmp:1]: Can we remove this .tmp file? This looks to just be MapUI.tscn

File: Map/MapBase.gd [hunk 2]
```diff
@@ -0,0 +1,9 @@
+extends Resource
+class_name MapBase
+## Class for the map. Provides functions for the map
+##
+## Holds basic functionality for a map
+
+@export var rooms: Array[Array] = [] ## 2D array of rooms on the map. this will be assigned when generating rooms
+@export var floors_width: Array[int] = [1, 3, 5, 7, 5, 3, 1] ## array for variable map width, each element is the width of a floor
```
Comment (Tysterman74) [inline Map/MapBase.gd:8]: Should we remove this?

**Review by Tysterman74 — DISMISSED** [review]

File: Map/MapBase.gd (hunk 2 above)
Comment (Turtyo) [reply Map/MapBase.gd:8]: that was a question I had in a previous PR, on map movement iirc
we can i think

File: #Scenes/Map31B5.tmp (hunk 1 above)
Comment (Turtyo) [reply #Scenes/Map31B5.tmp:1]: mmh I'm not sure, i'll remove it and see if nothing breaks

**Review by cheesefrycook — DISMISSED** [review]

**Review by cheesefrycook — APPROVED** [review]

**Pull request merged**